### PR TITLE
[bugfix] Use formatMessage ids and default messages per api

### PR DESCRIPTION
**Pull request by sirugh**
## Description

`formatMessage(obj)` takes an object like `{ id, defaultMessage, description }`. We were passing `defaultMessage` as `id`, which "worked" but only coincidentally. If you have some variable replacement within an `id`, there is a moment where the id will render as text before being replaced by the variable. See the below video.

Note: the issue from account chip is being fixed in #2672 

## Related Issue

Closes [PWA-913](https://jira.corp.magento.com/browse/PWA-913).

### Verification Steps

1. Check all the areas that are changed and ensure they still render the right messages.

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/Header/accountTrigger.js [hunk 1]
```diff
@@ -36,13 +36,17 @@ const AccountTrigger = props => {
             <div className={rootClassName} ref={accountMenuTriggerRef}>
                 <button
                     aria-label={formatMessage({
-                        id: 'Toggle My Account Menu'
+                        id: 'accountTrigger.button',
```
Comment (sirugh) [inline packages/venia-ui/lib/components/Header/accountTrigger.js:39]:
```suggestion
                        id: 'accountTrigger.ariaLabel',
```

File: packages/venia-ui/lib/components/Header/accountTrigger.js [hunk 2]
```diff
@@ -36,13 +36,17 @@ const AccountTrigger = props => {
             <div className={rootClassName} ref={accountMenuTriggerRef}>
                 <button
                     aria-label={formatMessage({
-                        id: 'Toggle My Account Menu'
+                        id: 'accountTrigger.ariaLabel',
```
[the remark this answers is not in the document]
Comment (anthoula) [reply packages/venia-ui/lib/components/Header/accountTrigger.js:39]: Keys should also be updated in: `packages/venia-ui/lib/i18n/en_US.json`

[the remark this answers is not in the document]
Comment (sirugh) [reply packages/venia-ui/lib/components/Header/accountTrigger.js:39]: omg I totally get it now - it's not that `formatMessage` renders the `id`, it's that we didn't provide a default message :) I can revert all this and just provide default messaging where we use variables.

**Comment by supernova-at** [comment]
Code looks good, will run verification steps when builds are finished.

Question though: instead of rendering a flash of string template content (`Hello, { name }`) like you have in the video, will this now render a flash of english content?

In other words, if we're in `FR-fr` locale we'll get a flash of "Hello, Stephen" and then "Bonjour, Stephen" ?

**Comment by sirugh** [comment]
> Question though: instead of rendering a flash of string template content (Hello, { name }) like you have in the video, will this now render a flash of english content (because the default values are english?

> In other words, if we're in FR-fr locale we'll get a flash of "Hello, Stephen" before the proper "Bonjour, Stephen" ?

Good call out Andy! I think you are correct, and the right place to prescribe defaults may be in the json, not the component.

**Review by anthoula — APPROVED** [review]

**Review by supernova-at — APPROVED** [review]

**Pull request merged**